libvirt-wrapper.py

```diff
@@ -0,0 +1,91 @@
+import libvirt
+import sys
+import argparse
+
+class LibvirtConn:
+    (ECONN, EFIND, ECREATE, EDELETE, ESTART, ESTOP, EREBOOT) = (1,2,3,4,5,6,7)
+    def __init__(self):
+        try:
+            self.conn = libvirt.open("qemu:///system")
+        except libvirt.libvirtError:
+            raise SystemExit(self.ECONN)
+
+    def find_vm_dec(func):
+        def closure(self, name, *args, **kwargs):
+                try:
+                    vm = self.conn.lookupByName(name)
+                except libvirt.libvirtError:
+                    raise SystemExit(self.EFIND)
+                return func(self, name, vm, *args, **kwargs)
+        return closure
+
+    def create_vm(self, xml_file):
+        with open(xml_file) as f:
+            xml = f.read()
+        try:
+            self.conn.defineXML(xml)
+        except libvirt.libvirtError:
+            raise SystemExit(self.ECREATE)
+
+    @find_vm_dec
+    def delete_vm(self, name, vm):
+        try:
+            if vm.isActive():
+                vm.destroy()
+            vm.undefine()
+        except libvirt.libvirtError:
+           raise SystemExit(self.EDELETE)
+
+    @find_vm_dec
+    def power_on(self, name, vm):
+        try:
+            vm.create()
+        except libvirt.libvirtError:
+            raise SystemExit(self.ESTART)
+
+    @find_vm_dec
+    def power_off(self, name, vm):
+        try:
+            vm.shutdown()
+        except libvirt.libvirtError:
+            raise SystemExit(self.ESTOP)
+
+    @find_vm_dec
+    def reboot(self, name, vm):
+        try:
+            vm.reboot()
+        except libvirt.libvirtError:
+            raise SystemExit(self.EREBOOT)
+
+
+parser = argparse.ArgumentParser(description="Pythonic wrapper for libvirt. "
```

Comment:
You might want to wrap the code beyond under `if __name__ == '__main__'`. Otherwise, if you'd decide to re-use your class and import it from other module, the argparser code will still be executed and most likely you don't want and don't expect that.